File: app/src/main/kotlin/ru/tutu/stations/ui/mvp/viewState/MvpViewState.java
package ru.tutu.stations.ui.mvp.viewState;

import ru.tutu.stations.ui.mvp.view.MvpView;

/**
 * Состояние {@link MvpView}.
 *
 * @param <V> {@link MvpView}
 * @author Grigoriy Pryamov
 */
public interface MvpViewState<V extends MvpView> {

    /**
     * Присоединение view.
     *
     * @param view Присоединенная view
     */
    void attachView(V view);

    /**
     * Отсоединение view.
     *
     * @param view Отсоединенная view
     */
    void detachView(V view);
}

File: app/src/main/kotlin/ru/tutu/stations/ui/mvp/viewstate/MvpViewState.kt
package ru.tutu.stations.ui.mvp.viewstate

import ru.tutu.stations.ui.mvp.MvpView

/**
 * Состояние {@link MvpView}.
 *
 * @param <V> {@link MvpView}
 * @author Grigoriy Pryamov.
 */
interface MvpViewState<V : MvpView> {
    /**
     * Присоединение view.
     *
     * @param view Присоединенная view
     */
    fun attachView(view: V)

    /**
     * Отсоединение view.
     *
     * @param view Отсоединенная view
     */
    fun detachView(view: V)
}